src/App.js

```diff
@@ -1,61 +1,87 @@
-  average_rating: 6,
-  genres: [{id: 18, name: "Drama"}, {id: 42, name: "Comedy"}],
-  budget: 63000000,
-  revenue: 100853753,
-  runtime: 139,
-  tagline: "It's a movie!"
-}
-
 class App extends Component {
+  _isLoaded = false
+
   constructor() {
     super();
+
     this.state = {
-      defaultMovie
+      allMovies: [],
+      currentMovie: this.getMovie(659986), // change this once onClick is fixed, to {}
+      error: ''
     }
   }
 
+  componentDidMount() {
+    this._isLoaded = true;
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies`)
+      .then(response => response.json())
+      .then(allMovies => this.setState({ allMovies: allMovies.movies }))
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  getMovie(id) {
+    this._isLoaded = true; // this isn't working correctly for getMovie, maybe because I'm calling it in the constructor
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+      .then(response => response.json())
+      .then(currentMovie => this.setState({ currentMovie: currentMovie.movie }))
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  // handleClick(id) {
+  //   // this._isLoaded = true; // this isn't working correctly for getMovie, maybe because I'm calling it in the constructor
+  //
+  //   fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+  //     .then(response => response.json())
+  //     // .then(data => console.log(data.movie))
+  //     .then(movieData => this.setState({ currentMovie: movieData.movie }))
+  //     .catch(error => this.setState({ error: error.message }))
+  // }
+
+  showAll() {
+    this.currentMovie = {}
+  }
+
+  componentWillUnmount() {
+    this._isLoaded = false;
+  }
+
+  // todo ==> refactor map over to its own component
   render() {
     return (
       <div className="App">
         <header>
           <img src={logo} className="App-logo" alt="logo" />
           <span className="title">Rancid Tomatillos</span>
-          {/*<section className="cards">*/}
-          {/*  {movieData.movies.map((movie) => {*/}
-          {/*    return <Movie*/}
-          {/*      key={movie.id}*/}
-          {/*      id={movie.id}*/}
-          {/*      poster_path={movie.poster_path}*/}
-          {/*      backdrop_path={movie.backdrop_path}*/}
-          {/*      title={movie.title}*/}
-          {/*      average_rating={movie.average_rating}*/}
-          {/*      release_date={movie.release_date}*/}
-          {/*    />*/}
-          {/*  })}*/}
-          {/*</section>*/}
-          {this.state.defaultMovie &&
-            <section className="profile">
-              <MovieProfile
-                key={defaultMovie.id}
-                data={defaultMovie}
+          <section className="cards">
+            {this.state.allMovies.map((movie) => {
```

Comment:
I'd also memoize this ("memoize" means "store in a variable to be referenced later") outside of the return, and then reference the variable in the return. It's always good to keep `map`s outside of your return statements.

Reply:
I was thinking I should probably refactor this to have a Movies component where map is done, similar to the way Ideas is mapped in its own component.

Reply:
Would that be a reasonable solution?